File: java/db/jpa/java-jpa-mysql/src/main/java/com/spring/jpa/mysql/domain/Board.java
package com.spring.jpa.mysql.domain;

import com.spring.jpa.core.domain.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;

@Entity
@Table(name = "board", catalog = "study")
@Builder
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE) // 👈 Builder와 함께 사용하기 위해 추가 (private 접근)
@Getter
public class Board extends BaseEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false)
    private String writer;   // 작성자 이름 또는 작성자 ID

    @Column(nullable = false)
    private int viewCount = 0;

    public void update(String title, String content){
        this.title = title;
        this.content = content;
    }
}

File: kotlin/db/jpa/kotlin-jpa-mysql/src/main/kotlin/com/spring/jpa/mysql/domain/Board.kt
package com.spring.jpa.mysql.domain

import com.spring.jpa.core.domain.entity.BaseEntity
import jakarta.persistence.*
import java.io.Serializable

@Entity
@Table(name = "board", catalog = "study")
data class Board(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false, length = 200)
    var title: String? = null,

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String? = null,

    @Column(nullable = false)
    var writer: String? = null,

    @Column
    var viewCount: Int? = 0
) : BaseEntity(), Serializable {
    fun update(title: String?, content: String?) {
        this.title = title
        this.content = content
    }
}
